src/lib/plugin_apis/lvm.api

```diff
@@ -270,6 +270,91 @@ GType bd_lvm_vgdata_get_type () {
     return type;
 }
 
+#define BD_LVM_TYPE_SEGDATA (bd_lvm_segdata_get_type ())
+GType bd_lvm_segdata_get_type();
+
+/**
+ * BDLVMSEGdata:
+ * @size_pe: Size of this segment in extents
+ * @pv_start_pe: Where it starts in the physical volume in extents
+ * @pvdev: Device name of the physical volume or NULL
+ */
+typedef struct BDLVMSEGdata {
+    guint64 size_pe;
+    guint64 pv_start_pe;
+    gchar *pvdev;
+} BDLVMSEGdata;
+
+/**
+ * bd_lvm_segdata_copy: (skip)
+ * @data: (nullable): %BDLVMSEGdata to copy
+ *
+ * Creates a new copy of @data.
+ */
+BDLVMSEGdata* bd_lvm_segdata_copy (BDLVMSEGdata *data) {
+    if (data == NULL)
+        return NULL;
+
+    BDLVMSEGdata *new_data = g_new0 (BDLVMSEGdata, 1);
+
+    new_data->size_pe = data->size_pe;
+    new_data->pv_start_pe = data->pv_start_pe;
+    new_data->pvdev = g_strdup (data->pvdev);
+    return new_data;
+}
+
+/**
+ * bd_lvm_segdata_free: (skip)
+ * @data: (nullable): %BDLVMSEGdata to free
+ *
+ * Frees @data.
+ */
+void bd_lvm_segdata_free (BDLVMSEGdata *data) {
+    if (data == NULL)
+        return;
+
+    g_free (data->pvdev);
+    g_free (data);
+}
+
+GType bd_lvm_segdata_get_type () {
+    static GType type = 0;
+
+    if (G_UNLIKELY(type == 0)) {
+        type = g_boxed_type_register_static("BDLVMSEGdata",
+                                            (GBoxedCopyFunc) bd_lvm_segdata_copy,
+                                            (GBoxedFreeFunc) bd_lvm_segdata_free)       ;
+    }
+
+    return type;
+}
+
+static BDLVMSEGdata **copy_segs (BDLVMSEGdata **segs) {
+    int len;
```

Comment:
Please use glib types here, looks like GIR scanner doesn't like int on 32bit:

```
/usr/lib/gcc/i686-linux-gnu/10/include/stddef.h:424: syntax error, unexpected identifier in '  __float128 __max_align_f128 __attribute__((__aligned__(__alignof(__float128))));' at '__float128'
/home/jenkins/workspace/libblockdev-pr/arch/i686/distro/debian_11/type/libblockdev/src/lib/plugin_apis/lvm.h:244: syntax error, unexpected BASIC_TYPE in '    for (int i = 0; i < len; i++)' at 'int'
/home/jenkins/workspace/libblockdev-pr/arch/i686/distro/debian_11/type/libblockdev/src/lib/plugin_apis/lvm.h:244: syntax error, unexpected identifier in '    for (int i = 0; i < len; i++)' at 'i'
/home/jenkins/workspace/libblockdev-pr/arch/i686/distro/debian_11/type/libblockdev/src/lib/plugin_apis/lvm.h:254: syntax error, unexpected BASIC_TYPE in '    for (int i = 0; segs[i]; i++)' at 'int'
/home/jenkins/workspace/libblockdev-pr/arch/i686/distro/debian_11/type/libblockdev/src/lib/plugin_apis/lvm.h:254: syntax error, unexpected identifier in '    for (int i = 0; segs[i]; i++)' at 'segs'
```

Reply:
Yes, seems to be have been `for (int i)`. Fixed.